app/modules/session/google-auth.server.ts

```diff
@@ -0,0 +1,13 @@
+import { OAuth2Client } from 'google-auth-library';
+
+const client = new OAuth2Client();
+
+/** Verify JWT token from Google authentication and return decoded payload. */
+export async function verifyGoogleToken(idToken: string) {
```

Comment:
This `verifyGoogleToken` function is in its own separate `google-auth.server.ts` module rather than being in the `action` function in the `/login` route. This is because calling `const client = new OAuth2Client();` triggers a `ReferenceError: process is not defined` if called in the `action` function, which seems to indicate that Remix thinks this code could be ran client-side.

Theorectorically, code in the `action` function should only be ran server-side, so I'm not sure why this error is happening. The workaround here is to have the code in a file with a `.server.ts` extension to indicate to Remix that the code will only run on the server.